test/integration/gardenlet/networkpolicy/networkpolicy_test.go

```diff
@@ -0,0 +1,156 @@
+		fooNamespace              *corev1.Namespace
+		kubernetesEndpoint        *corev1.Endpoints
+	)
+
+	BeforeEach(func() {
+		kubernetesEndpoint = &corev1.Endpoints{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: "default",
+				Name:      "kubernetes",
+			},
+		}
+
+		shootNamespace = &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "shoot--",
+				Labels:       map[string]string{v1beta1constants.GardenRole: v1beta1constants.GardenRoleShoot},
+			},
+		}
+
+		fooNamespace = &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "foo--",
+			},
+		}
+	})
+
+	JustBeforeEach(func() {
+		By("Construct expected network policy")
+		Expect(testClient.Get(ctx, client.ObjectKeyFromObject(kubernetesEndpoint), kubernetesEndpoint)).To(Succeed())
+
+		expectedNetworkPolicySpec = networkingv1.NetworkPolicySpec{
+			Egress: networkpolicyhelper.GetEgressRules(kubernetesEndpoint.Subsets...),
+			PodSelector: metav1.LabelSelector{
+				MatchLabels: map[string]string{"networking.gardener.cloud/to-seed-apiserver": "allowed"},
+			},
+			PolicyTypes: []networkingv1.PolicyType{"Egress"},
+		}
+
+		By("Create shoot namespace")
+		Expect(testClient.Create(ctx, shootNamespace)).To(Succeed())
+		log.Info("Created shoot namespace for test", "namespace", client.ObjectKeyFromObject(shootNamespace))
+
+		DeferCleanup(func() {
+			By("Delete shoot namespace")
+			Expect(client.IgnoreNotFound(testClient.Delete(ctx, shootNamespace))).To(Succeed())
+		})
+
+		By("Create foo namespace")
+		Expect(testClient.Create(ctx, fooNamespace)).To(Succeed())
+		log.Info("Created foo namespace for test", "namespace", client.ObjectKeyFromObject(fooNamespace))
+
+		DeferCleanup(func() {
+			By("Delete foo namespace")
+			Expect(client.IgnoreNotFound(testClient.Delete(ctx, fooNamespace))).To(Succeed())
+		})
+	})
+
+	Context("#Reconcile", func() {
+		It("should create the network policy in the shoot namespace", func() {
+			By("Wait for network policy controller creating the network policy")
+			Eventually(func(g Gomega) {
+				shootNetworkPolicy := &networkingv1.NetworkPolicy{}
+				g.Expect(testClient.Get(ctx, client.ObjectKey{Namespace: shootNamespace.Name, Name: "allow-to-seed-apiserver"}, shootNetworkPolicy)).To(Succeed())
+				g.Expect(shootNetworkPolicy.Spec).To(Equal(expectedNetworkPolicySpec))
+			}).Should(Succeed())
+		})
+
+		It("should reconcile the network policy in the shoot namespace when it is changed by a third party", func() {
+			By("Modify network policy in shoot namespace")
+			modifiedShootNetworkPolicy := &networkingv1.NetworkPolicy{}
+			Expect(testClient.Get(ctx, client.ObjectKey{Namespace: shootNamespace.Name, Name: "allow-to-seed-apiserver"}, modifiedShootNetworkPolicy)).To(Succeed())
+			modifiedShootNetworkPolicy.Spec.PodSelector.MatchLabels["foo"] = "bar"
+			Expect(testClient.Update(ctx, modifiedShootNetworkPolicy)).To(Succeed())
+
+			By("Wait for network policy controller reconciling the network policy")
```

Comment:
You can shorten this (same for similar occurrences):

```suggestion
			By("Wait for controller to reconcile the network policy")
```